examples/gps/gps_example.cc

```diff
@@ -0,0 +1,202 @@
+OSM_GPS_MAP_SOURCE_OSMC_TRAILS
+OSM_GPS_MAP_SOURCE_LAST
+*/
+
+
+// all interaction outside the gui has to be handled in these timers... yuck
+struct MapAndGps {
+    OsmGpsMap *map;
+    Gps *gps;
+
+    MapAndGps(OsmGpsMap *m, Gps *g) {
+        map = m;
+        gps = g;
+    }
+};
+
+gboolean on_timeout (gpointer user_data) {
+    MapAndGps *mgp = static_cast<MapAndGps*>(user_data);
+
+    units::angle::degree_t lat;
+    units::angle::degree_t lon;
+    units::length::meter_t alt;  
+
+    // get position from gps
+    mgp->gps->getPosition(lat,lon, alt);
+    
+    // update the map
+    osm_gps_map_set_center_and_zoom (mgp->map,
+                                     lat.value(),
+                                     lon.value(),
+                                     70);
+                                    
+  
+    return G_SOURCE_CONTINUE; /* or G_SOURCE_REMOVE when you want to stop */ 
+}
+
+int main(int argc, char** argv)
+{ 
+    GtkBuilder *builder;
+    GtkWidget *widget;
+    OsmGpsMap *map;
+    OsmGpsMapLayer *osd;
+
+    // options
+    static OsmGpsMapSource_t opt_map_provider = OSM_GPS_MAP_SOURCE_GOOGLE_SATELLITE;
+    static gboolean opt_friendly_cache = FALSE;
+    static gboolean opt_no_cache = FALSE;
+    static char *opt_cache_base_dir = NULL;
+    
+    
+    char *cachedir, *cachebasedir;
+    GError *error = NULL;
+    GOptionContext *context;
+
+    gtk_init (&argc, &argv);
+
+    context = g_option_context_new ("- Map browser");
+    g_option_context_set_help_enabled(context, FALSE);
+    cachebasedir = osm_gps_map_get_default_cache_directory();
+
+    if (opt_cache_base_dir && g_file_test(opt_cache_base_dir, G_FILE_TEST_IS_DIR)) {
+        cachedir = g_strdup(OSM_GPS_MAP_CACHE_AUTO);
+        cachebasedir = g_strdup(opt_cache_base_dir);
+    } else if (opt_friendly_cache) {
+        cachedir = g_strdup(OSM_GPS_MAP_CACHE_FRIENDLY);
+    } else if (opt_no_cache) {
+        cachedir = g_strdup(OSM_GPS_MAP_CACHE_DISABLED);
+    } else {
+        cachedir = g_strdup(OSM_GPS_MAP_CACHE_AUTO);
+    }
+    
+    map = (OsmGpsMap*) g_object_new (OSM_TYPE_GPS_MAP,
+                        "map-source",opt_map_provider,
+                        "tile-cache",cachedir,
+                        "tile-cache-base", cachebasedir,
+                        "proxy-uri",g_getenv("http_proxy"),
+                        "drag-limit",1,
+                        "auto-download", TRUE,
+                        "gps-track-point-radius", 1,
+                        NULL);
+
+    osd = (OsmGpsMapLayer*) g_object_new (OSM_TYPE_GPS_MAP_OSD,
+                        "show-scale",TRUE,
+                        "show-coordinates",TRUE,
+                        "show-crosshair",TRUE,
+                        "show-dpad",TRUE,
+                        "show-zoom",TRUE,
+                        "show-gps-in-dpad",TRUE,
+                        "show-gps-in-zoom",TRUE,
+                        "dpad-radius", 30,
+                        NULL);
```

Comment:
There is a really good C++ wrapper library for GTK: https://www.gtkmm.org/en/

This might be a better solution if we are planning on using GTK elsewhere.

Reply:
gtk is used by the mapping widget, otherwise I would never use this crappy library